include/envoy/server/instance.h

```diff
@@ -145,6 +145,11 @@ class Instance {
   virtual void shutdownAdmin() PURE;
 
   /**
+   * @return Singleton::Manager& the server-wide singleton manager.
+   */
+  virtual Singleton::Manager& singletonManager() PURE;
```

Comment:
High level question: should this subsume existing singleton state on the server? E.g. `initManager()`, `listenerManager()` etc.